File: core/actions/update-price.go
package actions

import "github.com/dc0d/reliable-messaging/core/model"

type UpdatePriceBuilder struct {
	Service model.EntityDomainService
}

func (builder UpdatePriceBuilder) Build() *UpdatePrice {
	result := &UpdatePrice{service: builder.Service}

	return result
}

type UpdatePrice struct {
	service model.EntityDomainService
}

func (action *UpdatePrice) Execute(incoming model.PriceUpdated) error {
	return action.service.UpdateStock(incoming)
}

File: stocks/src/core/ports/update-price.ts
import { StockDomainService } from "./../model/stock-domain-service";
import { PriceUpdated } from "./../model/events";

export class UpdatePrice {
  protected svc: StockDomainService;

  constructor(svc: StockDomainService) {
    this.svc = svc;
  }

  async execute(event: PriceUpdated): Promise<void> {
    await this.svc.updateStock(event);
  }
}
